From 1fbe470f76cf2b997121c0299f0bfb3698ae5e10 Mon Sep 17 00:00:00 2001
From: David Steele <dsteele@gmail.com>
Date: Wed, 8 Jan 2014 00:49:19 -0500
Subject: [PATCH 1/5] Disable exception processing in XMPP close()

When re-establising a lost connection, it was causing an exception
before the XMPP connection is attempted, requiring a restart to
recovery.

Introduced by 16dc1265
Closes davesteele/cloudprint-service#9
---
 cloudprint/xmpp.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/cloudprint/xmpp.py b/cloudprint/xmpp.py
index 64b1935..59231f4 100644
--- a/cloudprint/xmpp.py
+++ b/cloudprint/xmpp.py
@@ -142,12 +142,16 @@ class XmppConnection(object):
 
     def close(self):
         """Close the connection to the XMPP server"""
-        if self._wrappedsock is not None:
+        try:
             self._wrappedsock.shutdown(socket.SHUT_RDWR)
             self._wrappedsock.close()
+        except:
+            # close() is best effort. Don't respond to failures
+            LOGGER.debug("Error encountered closing XMPP socket")
+        finally:
+            self._connected = False
+            self._nextkeepalive = 0
             self._wrappedsock = None
-        self._connected = False
-        self._nextkeepalive = 0
 
 
     def is_connected(self):
-- 
1.8.5.2

